src/tests/JIT/opt/Subtract/IntSubtract.cs

```diff
@@ -0,0 +1,154 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace CodeGenTests
+{
+    static class IntSubtract
+    {
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static sbyte Int8_Subtract(sbyte x, sbyte y)
+        {
+            // X64-NOT: movsx
+
+            // X64:      sub
+            // X64-NEXT: movsx
+
+            return (sbyte)(x - y);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static byte UInt8_Subtract(byte x, byte y)
+        {
+            // X64-NOT: movzx
+
+            // X64:      sub
+            // X64-NEXT: movzx
+
+            return (byte)(x - y);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static short Int16_Subtract(short x, short y)
+        {
+            // X64-NOT: movsx
+
+            // X64:      sub
+            // X64-NEXT: movsx
+
+            // X64-NOT: cwde
+
+            return (short)(x - y);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static ushort UInt16_Subtract(ushort x, ushort y)
+        {
+            // X64-NOT: movzx
+
+            // X64:      sub
+            // X64-NEXT: movzx
+
+            return (ushort)(x - y);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static int Int32_Subtract(int x, int y)
+        {
+            // X64: mov
```

Comment:
I don't think there's a lot of value in checking for `mov` here.  It just a move.  It may also be adding some brittleness to the test.  Do you specifically want to allow
```
mov <ret>, <x>
sub <ret>, <y>
```
and not
```
sub <x>, <y>
mov <ret>, <x>
```
?

Reply:
I don't think this one will be considered brittle. Sure, this could be re-ordered, but I don't see that happening.

It may not be as valuable checking for Int32/Int64 subtraction in this case, but it's to be thorough considering we are checking the rest of the types.

Reply:
"I don't see that happening" seems rather speculative here.

What is the value in checking that we move first?  "thorough" doesn't mean to put a disasm check for every line of current disasm output - what are valuable checks that you could put here?  If it's just looking for `sub`, then fine.  If there's a reason why a compiler should emit a move first, then do `mov`/`sub`.  If it's CHECK-NOT movzx or whatever, then do that.

Extraneous lines have costs - the next person to look at the test will see "this test should emit a move before an add" and have to think about that.

Reply:
I can just use `X64-NOT: movzx` instead of just checking for `X64: mov` - that would at least check to make sure we don't do a zero/sign-extend for these types.